python/plugins/processing/modeler/ModelerDialog.py

```diff
@@ -486,7 +487,15 @@ def fillAlgorithmTreeUsingProviders(self):
                     continue
                 if alg.commandLineName() == self.alg.commandLineName():
                     continue
-                if text == '' or text.lower() in alg.name.lower():
+
+                item_text = [alg.name.lower()]
+                item_text.extend(alg.tags.split(','))
+
+                show = not search_strings or all(
```

Comment:
clear the result, hard (for me) to read the code. I'm not fan of pythonic syntax, and I would prefer explicit and readable code if no performance problems are involved. Is it possible to reformulate?